File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/battle/BattleManager.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.battle;

import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Player;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.action.EncounterState;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.action.PlayerEncounterAction;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Ship;

/**
 * Handles an encounter between the player and another ship.
 */
public abstract class BattleManager {
    final Player player;

    final Ship otherShip;

    /**
     * Creates an encounter between the ship and another ship.
     *
     * @param player the player
     */
    public BattleManager(Player player) {
        this.player = player;
        Ship playerShip = player.getShip();

        this.otherShip = Ship.getRandomShipWithWeapons();
    }

    /**
     * Starts the battle.
     *
     * @return a string detailing what is happening in the game
     */
    public abstract String startBattle();

    /**
     * Executes one turn for a given player action.
     *
     * @param pea the player's action
     * @param es whether the encounter is over or not
     * @return a string detailing what is happening in the game
     */
    public abstract String executeTurn(
            PlayerEncounterAction pea, EncounterState es);

    /**
     * @return the other ship in the encounter
     */
    public Ship getOtherShip() {
        return otherShip;
    }

    public String getOtherShipString() {
        return otherShip.toStringShipType();
    }

    public int getOtherShipHealth() {
        return otherShip.getHealth();
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/battle/BattleManager.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

class BattleManager {
    var player : Player
    var playerShip : Ship
    var otherShip : Ship
    
    init(player : Player) {
        self.player = player
        self.playerShip = player.ship!
        self.otherShip = Ship.getRandomShipWithWeapons()
    }
    
    func startBattle() -> String {
        // abstract
    }
    
    func executeTurn(pea : PlayerEncounterAction, es : EncounterState) -> String {
        // abstract
    }
    
    func getOtherShip() -> Ship {
        return self.otherShip
    }
    
}
